### PR TITLE
Feature/notification action

### PR DESCRIPTION
**Pull request by TheblueMan003**
Added Mute Notification Button for meetup
Added Notification Actions: Clicking on the notification open the app with the corresponding activity
Fix Chat Time

### CODE REVIEW

File: app/src/main/java/com/github/palFinderTeam/palfinder/meetups/activities/MeetUpView.kt [hunk 1]
```diff
@@ -181,7 +194,20 @@ class MeetUpView : AppCompatActivity() {
...
         }
     }
-
+    /**
+     * cannot mute/unmute a meetUp if you are not logged in
+     */
+    fun onMuteOrUnmute(v: View){
+        if(profileService.getLoggedInUserID() == null){
+            createPopUp(this,
+                { startActivity(Intent(this, LoginActivity::class.java)) },
```
Comment (zaclegarssure) [inline app/src/main/java/com/github/palFinderTeam/palfinder/meetups/activities/MeetUpView.kt:203]: Maybe we should do a helper for these recurrent popups (also if you want to test this don't hesitate to see how I did with the warning when joining a meetup with a blocked user)

File: app/src/main/java/com/github/palFinderTeam/palfinder/notification/NotificationHandler.kt [hunk 2]
```diff
@@ -14,6 +15,7 @@ import androidx.core.content.ContextCompat.getSystemService
 import com.github.palFinderTeam.palfinder.R
 import com.github.palFinderTeam.palfinder.cache.DictionaryCache
 import com.github.palFinderTeam.palfinder.cache.FileCache
+import com.github.palFinderTeam.palfinder.profile.USER_ID
```
Comment (zaclegarssure) [inline app/src/main/java/com/github/palFinderTeam/palfinder/notification/NotificationHandler.kt:18]: Unused import

File: app/src/main/java/com/github/palFinderTeam/palfinder/notification/NotificationHandler.kt [hunk 3]
```diff
@@ -56,11 +58,24 @@ class NotificationHandler (
     }
 
     @RequiresApi(Build.VERSION_CODES.O)
```
Comment (zaclegarssure) [inline app/src/main/java/com/github/palFinderTeam/palfinder/notification/NotificationHandler.kt:60]: Those SDK checks are apparently obsolete, you can safely remove them.

File: app/src/main/java/com/github/palFinderTeam/palfinder/profile/FirebaseProfileService.kt [hunk 4]
```diff
@@ -101,6 +102,40 @@ open class FirebaseProfileService @Inject constructor(
...
+    override suspend fun muteMeetup(user: ProfileUser, meetup: String): Response<Unit> {
+        return try {
+            if (!user.canMuteMeetup(meetup)) {
+                return Response.Failure("Cannot mute this meetup.")
+            }
+            val batch = db.batch()
+            batch.update(
+                db.collection(PROFILE_COLL).document(user.uuid),
+                MUTED_MEETUPS, FieldValue.arrayUnion(meetup)
+            )
```
Comment (zaclegarssure) [inline app/src/main/java/com/github/palFinderTeam/palfinder/profile/FirebaseProfileService.kt:114]: You don't have to use batch for this I think, the simple update methods with arrayUnions should be equivalent

File: app/src/main/java/com/github/palFinderTeam/palfinder/profile/FirebaseProfileService.kt [hunk 5]
```diff
@@ -101,6 +102,40 @@ open class FirebaseProfileService @Inject constructor(
...
+    override suspend fun unMuteMeetup(user: ProfileUser, meetup: String): Response<Unit> {
+        return try {
+            if (!user.canUnMuteMeetup(meetup)) {
+                return Response.Failure("Cannot unmute this meetup.")
+            }
+            val batch = db.batch()
+            batch.update(
+                db.collection(PROFILE_COLL).document(user.uuid),
+                MUTED_MEETUPS, FieldValue.arrayRemove(meetup)
+            )
```
Comment (zaclegarssure) [inline app/src/main/java/com/github/palFinderTeam/palfinder/profile/FirebaseProfileService.kt:131]: Same here

**Review by zaclegarssure — APPROVED** [review]
Other than that, it's' actually really cool and well done, nice job !

File: app/src/androidTest/java/com/github/palFinderTeam/palfinder/profile/UIMockProfileServiceModule.kt [hunk 6]
```diff
@@ -155,7 +157,7 @@ object UIMockProfileServiceModule {
         override suspend fun unfollowUser(user: ProfileUser, targetId: String): Response<Unit> {
             return try {
                 if (!user.canUnFollow(targetId)) {
-                    return Response.Failure("Cannot unfollow this user.")
+                    return Response.Failure("Cannot mute this meetup.")
```
Comment (atappy) [inline app/src/androidTest/java/com/github/palFinderTeam/palfinder/profile/UIMockProfileServiceModule.kt:160]: shouldn't we exchange the failure messages between unfollowUser and followUser ?

File: app/src/main/java/com/github/palFinderTeam/palfinder/utils/time/ShortDate.kt [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.icu.text.SimpleDateFormat
+import android.icu.util.Calendar
+import com.github.palFinderTeam.palfinder.R
+import java.util.*
+
+object ShortDate{
+    @SuppressLint("SimpleDateFormat")
```
Comment (atappy) [inline app/src/main/java/com/github/palFinderTeam/palfinder/utils/time/ShortDate.kt:11]: maybe add some comments ?

File: app/src/main/res/values/strings.xml [hunk 8]
```diff
@@ -135,6 +137,9 @@
     <string name="ok">OK</string>
 
     <string name="date_long_format">EEEE, MMMM d, yyyy \'at\' h:mm a</string>
+    <string name="date_very_short_format">h:mm</string>
+    <string name="date_short_format">MMMM d</string>
+    <string name="date_medium_format">MMMM d yyyy</string>
```
Comment (atappy) [inline app/src/main/res/values/strings.xml:142]: maybe change the date format to be consistent with the european point of view ?

File: app/src/test/java/com/github/palFinderTeam/palfinder/profile/MockProfileService.kt [hunk 9]
```diff
@@ -52,6 +52,31 @@ class MockProfileService : ProfileService {
...
+            Response.Success(Unit)
+        } catch (e: Exception) {
+            Response.Failure(e.message.orEmpty())
+        }
+    }
+
+    override suspend fun unMuteMeetup(user: ProfileUser, meetup: String): Response<Unit> {
+        return try {
+            if (!user.canUnMuteMeetup(meetup)) {
+                return Response.Failure("Cannot unfollow this user.")
```
Comment (atappy) [inline app/src/test/java/com/github/palFinderTeam/palfinder/profile/MockProfileService.kt:71]: same problem, we should maybe make it more consistent with the real utility of this function ?

File: app/src/main/java/com/github/palFinderTeam/palfinder/meetups/activities/MeetUpViewViewModel.kt [hunk 10]
```diff
@@ -126,6 +150,29 @@ class MeetUpViewViewModel @Inject constructor(
         }
     }
 
+    /**
+     * test if the user is participating in the meetup and propagate the join/leave to the database
```
Comment (atappy) [inline app/src/main/java/com/github/palFinderTeam/palfinder/meetups/activities/MeetUpViewViewModel.kt:154]: shouldn't it be 'propagate mute/unmute to the database' ?

**Review by atappy — APPROVED** [review]
besides small problems, everything is good

**Pull request merged**